core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -2448,7 +2448,7 @@ public final void clearAllSecondaryUserStores() {
     /**
      * {@inheritDoc}
      */
-    public final String[] getAllSecondaryRoles() throws UserStoreException {
+    public String[] getAllSecondaryRoles() throws UserStoreException {
```

Comment:
It is meaningful to have these method final in abstract class as we won't be modifying this method in other implementations on getting the roles from secondaries